[Ang Yang Cheng] iP

djiangel opened the pull request:
✨ **DukePro** ✨

- Text-based
- Easy to learn
- Fast to use

1. Add tasks
2. Delete tasks
3. Update tasks as done

Upcoming

- [ ] Managing deadlines
- [ ] Reminders

> _Free_ and _amazing_ product, download [here](https://github.com/djiangel/ip)

Some sample JAVA code for beginners, `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Commit:
Level-1

djiangel committed:
Level-1

djiangel committed:
Level-2

djiangel committed:
Level-3

djiangel committed:
Level-4

djiangel committed:
A-TextUiTesting

djiangel committed:
Level-4

djiangel committed:
Level-5

djiangel committed:
Level-6

djiangel committed:
branch-Level-7

djiangel committed:
branch-Level-8

djiangel committed:
Level-8

djiangel committed:
branch-Level-7

djiangel committed:
branch-Level-7

djiangel committed:
Merge branch 'branch-Level-7'

djiangel committed:
a

djiangel committed:
Level-8

djiangel committed:
branch-Level-7

djiangel committed:
Merge branch 'branch-Level-7'

djiangel committed:
A-MoreOOP

djiangel committed:
A-Packages

djiangel committed:
A-JUnit

djiangel committed:
branch-A-JavaDoc

djiangel committed:
branch-Level-9

chingh20 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,64 @@
+     * @return String list to make sense of user input
+     * @throws DukeException
+     */
+    public static String[] parse(String command) throws DukeException {
```
This method is quite long. Maybe you can consider extracting some lines out as different methods and call the methods, or declare the common used variable, such as fullCommand at the beginning of the method and return it at the end of the method :)

chingh20 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,135 @@
+     * @return List of task in file
+     * @throws IOException
+     */
+    public ArrayList<Task> load() throws IOException {
```
This method is also a bit long. Maybe you can extract some lines out into different methods, such as a method for checking the type of task to add or whether a task is done or not.

chingh20 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,135 @@
+                System.out.println(task);
+                String[] split = task.split(" , ");
+                String type = split[0];
+                boolean done;
```
Maybe naming this boolean as isDone or hasCompletedTask might be better? (since done may be read as a verb)

chingh20 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,92 @@
+    /**
+     * Method to run entire programme flow
+     */
+    public void run() {
```
This method is quite long. Maybe you can extract actions for each command as separate methods :)

chingh20 reviewed (commented):
I think your coding quality is quite good:) There's not a lot to comment about variable or methods names. Each line of your code is also quite short and clear too. Perhaps you can make several methods shorter to make it even better. Overall well done!!

sharmainec commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,104 @@
+     *
+     * @return the number of tasks in the list
+     */
+    public int size() {
```
Maybe we can use getSize() instead of size(), it will be easier to read this way.

sharmainec commented on src/main/java/duke/Todo.java:
```diff
@@ -0,0 +1,24 @@
+     * @param done
+     */
+    public Todo(String description, boolean done) {
+        super(description, done);
```
I think maybe renaming the boolean to isDone, instead of done, will make it more obvious that done is a boolean.

sharmainec reviewed (commented):
LGTM. Codes are easy to read! Maybe just some minor changes to make it even more readable?

djiangel committed:
Merge remote-tracking branch 'origin/add-gradle-support' into branch-A-Gradle

djiangel committed:
gradle

djiangel committed:
Level-10

djiangel committed:
A-Assertions

djiangel committed:
There are unchecked assumptions

These assumptions should be checked to prevent bugs

We use assert to ensure that these assumptions are true

If it fails, we can show an error message

djiangel committed:
Updated code quality
There are several breach of code quality initally

Updated certain method and variable names, updated javadocs for all methods

Helps to make the code more understandable and readable

djiangel committed:
Merge pull request #1 from djiangel/branch-A-Assertions

A-Assertions

djiangel committed:
Fixed merge conflics

djiangel committed:
Merge pull request #2 from djiangel/branch-A-CodeQuality

A-CodeQuality

djiangel committed:
added Ui.png

djiangel committed:
added user guide

djiangel committed:
User Guide

djiangel committed:
User Guide

djiangel committed:
User Guide

djiangel committed:
code quality changes
